src/services/dmr/DmrService.ts

```diff
@@ -179,17 +179,45 @@ export class DmrService {
     let msg: string = "An unspecified error has been received from the server";
     if (response.status !== 401) {
       let result: any = response.data;
-      if (result && result["failure-description"] != null) {
-        if (isNotNullOrUndefined(result["failure-description"]["domain-failure-description"])) {
-          msg = result["failure-description"]["domain-failure-description"];
-        } else {
-          msg = result["failure-description"];
+      if (isNotNullOrUndefined(result)) {
+        if (isNotNullOrUndefined(result["failure-description"])) {
+          if (isNotNullOrUndefined(result["failure-description"]["domain-failure-description"])) {
+            msg = result["failure-description"]["domain-failure-description"];
+          } else {
+            msg = result["failure-description"];
+          }
+        }
+        // see if we can find the cause of this error
+        let cause: string = this.findErrorCause(result);
+        if (isNotNullOrUndefined(cause)) {
+          msg = cause;
         }
       }
       promise.reject(msg);
     }
   }
 
+  private findErrorCause(root: any): string {
+    // traverse the object tree
+    let cause: string = undefined;
+    traverseObject(root, (key: string, value: any, trail: string []) => {
```

Comment:
Can be simplified to `traverseObject(root, (key, value, trail) => {`

Also, as you are using `traverseObject` does that not mean that it's possible for the cause to be set multiple times with the current code? Don't we just want to return after finding the first cause?

Reply:
Yes, we do. It does not matter much tbh. We don't have abort the loop mechanism in traverseObject...how would you abort looping?

Reply:
You could modify `traverseObject` so that it accepts a callback which returns true once the desired state has been reached. If the callback returns false, keep traversing, otherwise stop.